src/cmd/link/internal/wasm/asm.go

```diff
@@ -410,8 +410,9 @@ func writeMemorySec(ctxt *ld.Link, ldr *loader.Loader) {
 	const wasmPageSize = 64 << 10 // 64KB
 
 	writeUleb128(ctxt.Out, 1)                        // number of memories
-	ctxt.Out.WriteByte(0x00)                         // no maximum memory size
+	ctxt.Out.WriteByte(0x01)                         // should be 0x01, when maximum memory needs to be set; otherwise, 0x00
 	writeUleb128(ctxt.Out, initialSize/wasmPageSize) // minimum (initial) memory size
+	writeUleb128(ctxt.Out, 1024)                     // maximum (initial) memory size
```

Comment:
Is there a way to make sure mininum is indeed less than maximum? Like assert or comments if that is obvious.

Reply:
Good point. panic when that happens, ref https://github.com/ethstorage/go/pull/3/commits/c1b1a0e7032fc5c2378b23a083131d4bd4f131e9